micronota/bfillings/infernal.py

```diff
@@ -0,0 +1,239 @@
+        # turn all heuristic filters off (slow)
+        '--max',
+        # skip all HMM filter stages, use only CM (slow)
+        '--nohmm',
+        # skip first two HMM filter stages (SSV & Vit)
+        '--mid',
+        # default: run search space size-dependent pipeline  [default]
+        '--default',
+        # set heuristic filters at Rfam-level (fast)
+        '--rfam',
+        # use HMM only, don't use a CM at all
+        '--hmmonly',
+        # do not allow truncated hits at sequence termini
+        '--notrunc',
+        # allow truncated hits anywhere within sequences
+        '--anytrunc',
+        # turn off the NULL3 post hoc additional null model
+        '--nonull3',
+        # use scanning CM CYK algorithm, not Inside in final stage
+        '--cyk',
+        # align hits with CYK, not optimal accuracy
+        '--acyk',
+        # only search the top strand
+        '--toponly',
+        # only search the bottom strand
+        '--bottomonly'
+    ]
+    _parameters = {}
+    _parameters.update({
+        i: _get_parameter(
+            ValuedParameter, i, Delimiter=' ', IsPath=True)
+        for i in _valued_path_options})
+    _parameters.update({
+        i: _get_parameter(
+            ValuedParameter, i, Delimiter=' ')
+        for i in _valued_nonpath_options})
+    _parameters.update({
+        i: _get_parameter(FlagParameter, i)
+        for i in _flag_options})
+
+    def _accept_exit_status(self, exit_status):
+        return exit_status == 0
+
+    def _get_result_paths(self, data):
+        result = {}
+        for i in self._valued_path_options:
+            o = self.Parameters[i]
+            if o.isOn():
+                out_fp = self._absolute(o.Value)
+                result[i] = ResultPath(Path=out_fp, IsWritten=True)
+        return result
+
+
+class CMFetch(CommandLineApplication):
+    '''cmfetch application controller.
+
+    cmfetch is used to get a covariance model by name or accession
+    from a CM database.
+    This wrapper is tested for INFERNAL 1.1.1 (July 2014).
+    '''
+    pass
+
+
+class CMPress(CommandLineApplication):
+    '''cmpress application controller.
+
+    cmpress is used to format a CM database into a binary format for cmscan.
+    This wrapper is tested for INFERNAL 1.1.1 (July 2014).
+    '''
+    _suppress_stderr = False
+    _command = 'cmpress'
+    _parameters = {
+        # force overwrite
+        '-F': FlagParameter(Prefix='-', Name='F')}
+
+    def _accept_exit_status(self, exit_status):
+        return exit_status == 0
+
+
+def cmpress_cm(cm, force=False):
+    '''Compress the CM database.
+
+    Parameters
+    ----------
+    cm : str
+        The file path to CM database.
+    force : boolean
+        Whether to overwrite.'''
+
+    app = CMPress(InputHandler='_input_as_path')
+    if force is True:
+        app.Parameters['-F'].on()
+    res = app(cm)
+    return res
+
+
+def cmscan_fasta(cm, in_fp, out_fp, evalue=0.01, cores=0, params=None):
```

Comment:
Yeah, we can check for data types on run time

This is done in [pwmantel](https://github.com/biocore/scikit-bio/blob/master/skbio/stats/distance/_mantel.py) in skbio